### PR TITLE
OSDOCS-6970 Adding NatGateway outbound type

### PR DESCRIPTION
**Pull request by bscott-rh**
Version(s):
4.14

Issue:
https://issues.redhat.com/browse/OSDOCS-6970

Link to docs preview:
[Installation configuration parameters](https://63181--docspreview.netlify.app/openshift-enterprise/latest/installing/installing_azure/installation-config-parameters-azure#installation-configuration-parameters-additional-azure_installation-config-parameters-azure)
[Required permissions for Azure IPI](https://63181--docspreview.netlify.app/openshift-enterprise/latest/installing/installing_azure/installing-azure-account#minimum-required-permissions-ipi-azure_installing-azure-account)

QE review:
- [x] QE has approved this change.

### CODE REVIEW

**Comment by bscott-rh** [comment]
@r4f4 @jhixson74 Hello Rafael and John, I have a couple of questions regarding the nat gateway outbound type documentation. Is there any additional configuration information that we need to convey in the docs besides simply setting the `outboundType` parameter correctly? Also, will this change apply to azure stack hub as well as regular azure? Thanks

**Comment by r4f4** [comment]
> @r4f4 @jhixson74 Hello Rafael and John, I have a couple of questions regarding the nat gateway outbound type documentation. Is there any additional configuration information that we need to convey in the docs besides simply setting the `outboundType` parameter correctly? Also, will this change apply to azure stack hub as well as regular azure? Thanks

I think it would be good to mention that the Installer will create one NAT gateway for each availability zone specified.
No StackHub support yet, only Public Cloud.

**Comment by bscott-rh** [comment]
> I think it would be good to mention that the Installer will create one NAT gateway for each availability zone specified. No StackHub support yet, only Public Cloud.

Thanks, updated the PR to include this. PTAL

**Comment by r4f4** [comment]
LGTM but let's wait a bit to see if we'll have to mention anything about the control plane machineset manifests.

File: modules/installation-configuration-parameters.adoc [hunk 1]
```diff
@@ -1320,8 +1320,8 @@ Additional Azure configuration parameters are described in the following table:
 you are using user-defined routing, you must have pre-existing networking
 available where the outbound routing has already been configured prior to
 installing a cluster. The installation program is not responsible for
-configuring user-defined routing.
-|`LoadBalancer` or `UserDefinedRouting`. The default is `LoadBalancer`.
+configuring user-defined routing. If you specify the `NatGateway` routing strategy, the installation program will create one NAT gateway for each availability zone that you specify in your `install-config.yaml` configuration file. The `NatGateway` outbound routing strategy is a Technology Preview feature in {product-name} {product-version}.
```
Comment (jeana-redhat) [inline modules/installation-configuration-parameters.adoc:1323]: I stole your text and therefore noticed:
```suggestion
configuring user-defined routing. If you specify the `NatGateway` routing strategy, the installation program will create one NAT gateway for each availability zone that you specify in your `install-config.yaml` configuration file. The `NatGateway` outbound routing strategy is a Technology Preview feature in {product-title} {product-version}.
```

**Comment by jinyunma** [comment]
I tested to install OCP cluster with NatGateway outboundtype under minimal permissions against 4.14 nightly build, found that two additional permissions are required:
 -  "Microsoft.Network/natGateways/read",
 -  "Microsoft.Network/natGateways/write"
 Suggestion to also mention this part in https://docs.openshift.com/container-platform/4.13/installing/installing_azure/installing-azure-account.html

**Comment by bscott-rh** [comment]
@r4f4 @jinyunma Thank you for the input. I've updated the PR with the permissions information and single-zone wording. PTAL

**Review by r4f4 — COMMENTED** [review]
/lgtm

File: modules/installation-configuration-parameters.adoc [hunk 2]
```diff
@@ -1116,8 +1116,8 @@ within link:https://azure.microsoft.com/en-us/global-infrastructure/regions[a re
 you are using user-defined routing, you must have pre-existing networking
 available where the outbound routing has already been configured prior to
 installing a cluster. The installation program is not responsible for
-configuring user-defined routing.
-|`LoadBalancer` or `UserDefinedRouting`. The default is `LoadBalancer`.
+configuring user-defined routing. If you specify the `NatGateway` routing strategy, the installation program will only create one NAT gateway. If you specify the `NatGateway` routing strategy, your account must have the `Microsoft.Network/natGateways/read` and `Microsoft.Network/natGateways/write` permissions. The `NatGateway` outbound routing strategy is a Technology Preview feature in {product-title} {product-version}.
```
Comment (kcarmichael08) [inline modules/installation-configuration-parameters.adoc:1119]:
```suggestion
configuring user-defined routing. If you specify the `NatGateway` routing strategy, the installation program will create only one NAT gateway. If you specify the `NatGateway` routing strategy, your account must have the `Microsoft.Network/natGateways/read` and `Microsoft.Network/natGateways/write` permissions. The `NatGateway` outbound routing strategy is a Technology Preview feature in {product-title} {product-version}.
```

Comment (kcarmichael08) [reply modules/installation-configuration-parameters.adoc:1119]: Do we need the Technology Preview snippet to be included, like in a footnote, or is it OK to mention a TP feature without having the snippet?

**Review by kcarmichael08 — COMMENTED** [review]
I had a minor suggestion & question; otherwise lgtm!

File: modules/installation-configuration-parameters.adoc (hunk 2 above)
Comment (bscott-rh) [reply modules/installation-configuration-parameters.adoc:1119]: Let me try including the snippet in the table and see if it messes with the formatting.

Comment (kcarmichael08) [reply modules/installation-configuration-parameters.adoc:1119]: Yeah, it might - good luck <fingers crossed>

Comment (bscott-rh) [reply modules/installation-configuration-parameters.adoc]: Looks OK in my local build - PTAL once travis and netlify update in a few minutes. Thanks

Comment (kcarmichael08) [reply modules/installation-configuration-parameters.adoc]: It looks probably as good as it's going to, which is to say not awesome, but it's not breaking the table at least. :)

**Pull request merged**